### PR TITLE
feat: map interaction and tree hover/click

### PR DESCRIPTION
**Pull request by Jaszkowic**
- map interaction hook enables hover/click feature
- tree circle styling
- treeId as url search param on click (in preparation for tree detail view)

### CODE REVIEW

File: src/components/map/hooks/use-map-interaction.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    hoveredTreeIdRef.current = hoveredTreeId;
+  }, [hoveredTreeId]);
+
+  const [selectedTreeId, setSelectedTreeId] = useState<string | undefined>(
+    undefined,
+  );
+  const selectedTreeIdRef = useRef<string | undefined>(undefined);
+  useEffect(() => {
+    selectedTreeIdRef.current = selectedTreeId;
+  }, [selectedTreeId]);
```
Comment (raphael-arce) [inline src/components/map/hooks/use-map-interaction.tsx:25]: suggestion/comment: I find using plain `useEffect` always a bit hard to read/understand. What do you think about extracting line 19-25 into a named function `useSelectedTreeId` which just returns the ref? Same goes for the `hoveredTreeId` line 11-17

File: src/components/navbar/navbar.tsx [hunk 2]
```diff
@@ -35,6 +36,7 @@ const Navbar: React.FC = () => {
             onClick={(e) => {
               e.preventDefault();
               setPathname(item.path);
+              setSearchParams(new URLSearchParams());
```
[the remark this answers is not in the document]
Comment (Jaszkowic) [reply src/components/navbar/navbar.tsx:39]: yes, i think you are right. i don't see a use case for keeping search params. will change

File: src/components/map/hooks/use-map-constants.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import resolveConfig from "tailwindcss/resolveConfig";
+//@ts-ignore
+import tailwindConfig from "../../../../tailwind.config.js";
+const fullConfig = resolveConfig(tailwindConfig);
+
+export function useMapConstants() {
+  const MAP_PITCH_DEGREES = 45;
+  const ZOOMED_IN_ZOOM_LEVEL = 20;
```
[the remark this answers is not in the document]
Comment (Jaszkowic) [reply src/components/map/hooks/use-map-constants.tsx:8]: legacy variable name. i'll change it to `MAX_ZOOM_LEVEL` which describes the zoom level when you click on a tree and it zooms in.

File: src/components/map/hooks/use-map-interaction.tsx (hunk 1 above)
Comment (Jaszkowic) [reply src/components/map/hooks/use-map-interaction.tsx]: good idea. done

File: src/components/map/hooks/use-map-constants.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import resolveConfig from "tailwindcss/resolveConfig";
+//@ts-ignore
+import tailwindConfig from "../../../../tailwind.config.js";
+const fullConfig = resolveConfig(tailwindConfig);
+
+export function useMapConstants() {
```
[the remark this answers is not in the document]
Comment (Jaszkowic) [reply src/components/map/hooks/use-map-constants.tsx:6]: doesn't hurt to have them in env! i changed it.

**Review by raphael-arce — APPROVED** [review]
the preview is broken, idk why. but otherwise lgtm! 👍

**Comment by Jaszkowic** [comment]
.env were missing in Vercel, that's why it was broken. i fixed it

**Pull request merged**